core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java

```diff
@@ -178,13 +178,10 @@ public final synchronized RunT newBuild() throws IOException {
             return lastBuild;
         } catch (InstantiationException e) {
             throw new Error(e);
-        } catch (IllegalAccessException e) {
-            throw new Error(e);
-        } catch (InvocationTargetException e) {
-            throw handleInvocationTargetException(e);
-        } catch (NoSuchMethodException e) {
+        } catch (ReflectiveOperationException e) {
+            LOGGER.log(Level.WARNING, String.format("A new build could not be created in job %s", asJob().getFullName()), e);
```

Comment:
`InstantiationException` is also a ReflectiveOperationException, and in the case that is thrown you have no logging.